src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -361,29 +328,29 @@
     (let [population-registry (get-in metrics/metrics-registries [:population :registry])]
       (pop/initialize-population-metrics! population-registry read-db))
 
-    (when (.exists discard-dir)
-      (dlo/create-metrics-for-dlo! discard-dir))
-    (let [broker (try
-                   (log/info "Starting broker")
-                   (mq/build-and-start-broker! "localhost"
-                                               (conf/mq-dir config)
-                                               command-processing)
-                   (catch java.io.EOFException e
-                     (log/error
-                      "EOF Exception caught during broker start, this "
-                      "might be due to KahaDB corruption. Consult the "
-                      "PuppetDB troubleshooting guide.")
-                     (throw e)))
+    ;; Error handling here?
+    (let [stockdir (conf/stockpile-dir config)
+          command-chan (async/chan
+                         (queue/sorted-command-buffer
+                          max-enqueued
+                          #(mql/update-counter! :invalidated %1 %2 inc!)))
+          [q load-messages] (queue/create-or-open-stockpile (conf/stockpile-dir config))
           globals {:scf-read-db read-db
                    :scf-write-db write-db
-                   :pretty-print pretty-print}
-          clean-lock (ReentrantLock.)]
-      (transfer-old-messages! (conf/mq-endpoint config))
+                   :pretty-print pretty-print
+                   :q q
+                   :dlo (dlo/initialize (get-path stockdir "discard")
+                                        (get-in metrics-registries
+                                                [:dlo :registry]))
+                   :command-chan command-chan}
+          clean-lock (ReentrantLock.)
+          command-loader (when load-messages
```

Comment:
Not sure this is an issue, but I wondered about what happens if any of the remaining code in this function throws an exception after we start the future, i.e. does the tk lifecycle say that stop will be called, or do we know that the loader will behave reasonably regardless?